### ISSUE TITLE
Supergluu new user registration failing / Android 7.1.1

### ISSUE THREAD

**Issue by mzico** [GluuFederation/android-super-gluu#43]
Gluu Server version: 4.1.0 final
Super Gluu version: 3.1.3 
Android: 7.1.1 / Lenovo Tablet 

Here is the recorded situation: 
[Test.zip](https://github.com/GluuFederation/android-super-gluu/files/4880438/Test.zip)

And, this is the log I see in oxauth_script.log: 

```
2020-07-06 17:24:44,289 DEBUG [qtp1590550415-18] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:277) - Executing python 'getPageForStep' authenticator method
2020-07-06 17:24:45,308 DEBUG [qtp1590550415-15] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:384) - Validating acr_values: 'super_gluu'
2020-07-06 17:24:45,308 DEBUG [qtp1590550415-15] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:151) - Executing python 'isValidAuthenticationMethod' authenticator method
2020-07-06 17:24:45,309 DEBUG [qtp1590550415-15] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:249) - Executing python 'prepareForStep' authenticator method
2020-07-06 17:24:45,310 INFO  [qtp1590550415-15] [org.gluu.service.PythonService$PythonLoggerOutputStream] (PythonService.java:240) - Super-Gluu. Prepare for step 1
2020-07-06 17:24:45,311 DEBUG [qtp1590550415-15] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:263) - Executing python 'getExtraParametersForStep' authenticator method
2020-07-06 17:24:57,715 DEBUG [qtp1590550415-11] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:193) - Executing python 'authenticate' authenticator method
2020-07-06 17:24:57,719 INFO  [qtp1590550415-11] [org.gluu.service.PythonService$PythonLoggerOutputStream] (PythonService.java:240) - Super-Gluu. Authenticate for step 1
2020-07-06 17:24:57,741 INFO  [qtp1590550415-11] [org.gluu.service.PythonService$PythonLoggerOutputStream] (PythonService.java:240) - Super-Gluu. Authenticate for step 1. There is no U2F 'zico' user devices associated with application 'https://test41.gluu.org/identity/authcode.htm'. Changing auth_method to 'enroll'
2020-07-06 17:24:57,741 INFO  [qtp1590550415-11] [org.gluu.service.PythonService$PythonLoggerOutputStream] (PythonService.java:240) - Super-Gluu. Authenticate for step 1. auth_method: 'enroll'
2020-07-06 17:24:57,741 DEBUG [qtp1590550415-11] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:291) - Executing python 'getApiVersion' authenticator method
2020-07-06 17:24:57,741 DEBUG [qtp1590550415-11] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:207) - Executing python 'getNextStep' authenticator method
2020-07-06 17:24:57,742 DEBUG [qtp1590550415-11] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:263) - Executing python 'getExtraParametersForStep' authenticator method
2020-07-06 17:24:57,742 DEBUG [qtp1590550415-11] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:179) - Executing python 'getCountAuthenticationSteps' authenticator method
2020-07-06 17:24:57,745 DEBUG [qtp1590550415-11] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:277) - Executing python 'getPageForStep' authenticator method
2020-07-06 17:24:57,746 INFO  [qtp1590550415-11] [org.gluu.service.PythonService$PythonLoggerOutputStream] (PythonService.java:240) - Super-Gluu. authmethod 'enroll'
2020-07-06 17:24:57,746 DEBUG [qtp1590550415-11] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:263) - Executing python 'getExtraParametersForStep' authenticator method
2020-07-06 17:24:58,764 DEBUG [qtp1590550415-11] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:384) - Validating acr_values: 'super_gluu'
2020-07-06 17:24:58,765 DEBUG [qtp1590550415-11] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:151) - Executing python 'isValidAuthenticationMethod' authenticator method
2020-07-06 17:24:58,765 DEBUG [qtp1590550415-11] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:249) - Executing python 'prepareForStep' authenticator method
2020-07-06 17:24:58,766 INFO  [qtp1590550415-11] [org.gluu.service.PythonService$PythonLoggerOutputStream] (PythonService.java:240) - Super-Gluu. Prepare for step 2
2020-07-06 17:24:58,777 INFO  [qtp1590550415-11] [org.gluu.service.PythonService$PythonLoggerOutputStream] (PythonService.java:240) - Super-Gluu. Prepare for step 2. auth_method: 'enroll'
2020-07-06 17:24:58,779 INFO  [qtp1590550415-11] [org.gluu.service.PythonService$PythonLoggerOutputStream] (PythonService.java:240) - Super-Gluu. Prepare for step 2. Adding req_ip and req_loc to super_gluu_request
2020-07-06 17:24:58,779 INFO  [qtp1590550415-11] [org.gluu.service.PythonService$PythonLoggerOutputStream] (PythonService.java:240) - Super-Gluu. Determine remote location. remote_ip: '103.19.130.250'
2020-07-06 17:24:58,874 INFO  [qtp1590550415-11] [org.gluu.service.PythonService$PythonLoggerOutputStream] (PythonService.java:240) - Super-Gluu. Prepare for step 2. Prepared super_gluu_request:
2020-07-06 17:24:58,874 INFO  [qtp1590550415-11] [org.gluu.service.PythonService$PythonLoggerOutputStream] (PythonService.java:240) -  {"app":"https://test41.gluu.org/identity/authcode.htm","licensed":false,"method":"enroll","req_ip":"xxxx","created":"2020-07-06T17:24:58Z","issuer":"https://test41.gluu.org","req_loc":"xxxx%2C%2xxxx%20xxxx%2C%20xxxx","state":"2ed4fa83-97d7-438e-aecd-ab9c62d8b661","username":"zico"}
2020-07-06 17:24:58,875 DEBUG [qtp1590550415-11] [org.gluu.oxauth.service.external.ExternalAuthenticationService] (ExternalAuthenticationService.java:263) - Executing python 'getExtraParametersForStep' authenticator method
```

**Comment by mzico** [comment]
It's happening in 4.1 and 4.2

**Comment by mzico** [comment]
@soulfoodz : any update?

**Comment by crazedmeph** [comment]
@mzico @soulfoodz  This is still happening to me using version superglu v4.3.2 the key is not getting stored in superglu. If it is a brand new user it works successfully but does not store the key within superglu. Then once you try to login again it gives invalid token again.

**Comment by nynymike** [comment]
We moved this project to a monorepo https://github.com/GluuFederation/super-gluu 
We are pusing a new version of Super Gluu with fixes for 4.3.2.
